doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid mistakes of the detector. The last four float numbers we've got denote
+normalized coordinates and are between 0 and 1; to get the real pixel
+coordinates we should multiply it by the image sizes respectively to the
+directions.
+
+Then, to create masks we need to get landmarks of facial elements for every
+detected face. The chosen detector expects to obtain square area with only one
+face contained so we have to infer the model separately for every ROI we've
+got from the previous detection. The output of the facial landmarks detector is
+a blob with 35 landmarks: the first 18 of them are facial elements
+(eyes, eyebrows, a nose, a mouth) and the last 17 --- a jaw contour. Landmarks
+are floating point values of coordinates normalized relatively to an input ROI
+(not the original frame). In addition, for the further goals we need contours of
+eyes, mouths, faces, etc., not the landmarks. So, post-processing of the Mat is
+also required here. The process is split into two parts --- landmarks'
+coordinates denormalization to the real pixel coordinates of the source frame
+and getting necessary closed contours based on these coordinates.
+
+The last step of processing the inference data is drawing masks using the
+received contours. Honestly speaking, there is no need to get the most accurate
+contours in the world: we can always blur masks borders by gaussian filter to
+occupy a bit more area. Another point that should be mentioned here is getting
+three masks (for areas to be smoothed, for ones to be sharpened and for the
+background) which have no intersections with each other; this approach allows to
+apply the calculated masks to the corresponding images prepared beforehand and
+then just to summarize them to get the output image without any other actions.
+
+All the operations described above are supported fascinatingly by the G-API
+functionality, so this algorithm is appropriate to illustrate G-API usage
+convenience and efficiency in the context of solving a real CV/DL problem.
+
+# Preparing networks to be inferenced with G-API {#gapi_fb_prep_nets}
+
+Let's prepare our chosen networks for being processed inside the graph.
+Providing a new network support is quite similar to a new operation
+implementation.
+
+To declare a deep learning topologies we should use the specific macro
+`G_API_NET()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_decl
+
+Similarly to `G_API_OP()` macro for a new operation, it takes three arguments to
+register a new type.
+They are:
+1. Network interface name --- also serves as a name of a new type defined
+   with this macro;
+2. Network signature      --- an `std::function<>`-like signature which defines
+   inputs and outputs;
+3. Network's unique name  --- used to identify it within the system when the
+   type is stripped.
```

Comment:
What means `stripped` here? This is an implementation detail taken from internal comments.

I'd just refer to the "face analytics" tutorial here.